lib/reindex/s3_bucketd.py

```diff
@@ -350,3 +356,26 @@ def log_report(resource, name, obj_count, total_size):
             update_redis(pipeline, 'accounts', userid, report['obj_count'], report['total_size'])
             log_report('accounts', userid, report['obj_count'], report['total_size'])
         pipeline.execute()
+
+    observed_accounts = set(account_reports.keys())
+    recorded_accounts = set(get_resources_from_redis(redis_client, 'accounts'))
+    recorded_buckets = set(get_resources_from_redis(redis_client, 'buckets'))
+
+    # Stale accounts and buckets are ones that do not appear in the listing, but have recorded values
+    stale_accounts = recorded_accounts.difference(observed_accounts)
+    _log.info('Found %s stale accounts' % len(stale_accounts))
+    for chunk in chunks(stale_accounts, ACCOUNT_UPDATE_CHUNKSIZE):
+        pipeline = redis_client.pipeline(transaction=False) # No transaction to reduce redis load
+        for account in chunk:
+            update_redis(pipeline, 'accounts', account, 0, 0)
+            log_report('accounts', account, 0, 0)
+        pipeline.execute()
+
+    stale_buckets = recorded_buckets.difference(observed_buckets)
```

Comment:
I was unaware this exists in python (`difference`). neat.

Reply:
I think you can also use the `-` operator but tbh the named method works better.